### PR TITLE
Fix post upload notification messaging and analytics tracking

### LINKED ISSUE #6272: Media: improve upload notification messages
Originally coming [from this comment](https://github.com/wordpress-mobile/WordPress-Android/pull/6233#issuecomment-314249133) by @aforcier 

The notification when the upload completes has the wrong message in some cases.

1. We don't get 'Draft uploaded' messages when a post is saved as a remote draft. I think this is because we shouldn't run [this check](https://github.com/wordpress-mobile/WordPress-Android/blob/98b90c1a30813bbe0f7ea900d3d7e88b31de773d/WordPress/src/main/java/org/wordpress/android/ui/posts/services/PostUploadNotifier.java#L120) if the post is a `DRAFT`, as `mIsFirstTimePublish == false` always for remote drafts.
2. We're not flagging first time publishing (and thus also not tracking analytics) at all when uploading from the post list - we should do a check [here](https://github.com/wordpress-mobile/WordPress-Android/blob/98b90c1a30813bbe0f7ea900d3d7e88b31de773d/WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListFragment.java#L585) and use `PostUploadService.addPostToUploadAndTrackAnalytics()` instead where applicable

### PR DESCRIPTION
**Pull request by aforcier**
Fixes #6272.

There were three issues addressed:
1. The `PostUploadNotifier` wasn't displaying 'Draft uploaded' when uploading a draft post
2. The post list was not doing a 'first time publish' check (this also means we weren't tracking `EDITOR_PUBLISHED_POST` events when uploading posts from the first time from the post list)
3. We seem to be no longer initializing new posts with a default status, so the [`isFirstTimePublish()`](https://github.com/wordpress-mobile/WordPress-Android/blob/787a8e271d2456fc9ca1d081d66711bd6ee7ed39/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java#L1215) check was returning false negatives for new posts, also affecting messaging and analytics tracking

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -308,6 +308,7 @@ protected void onCreate(Bundle savedInstanceState) {
                     postFormat = SiteSettingsInterface.getDefaultFormat(WordPress.getContext());
                 }
                 mPost = mPostStore.instantiatePostModel(mSite, mIsPage, categories, postFormat);
+                mPost.setStatus(PostStatus.PUBLISHED.toString());
```
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: wondering - shouldn't the default be `DRAFT` for a new post?

Comment (aforcier) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Our default status has always been `PUBLISHED` - it's the status of the post if you write a new post and hit 'Publish' without changing anything. `DRAFT` is actually a special case. `Publish` is also what you would see as the default setting if you opened the post settings screen.

Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: gotcha - thanks!

**Comment by mzorz** [comment]
:shipit:

**Pull request merged**